### ISSUE TITLE
[BUG] Bars flickering with "Link Action Bars" enabled

### ISSUE THREAD

**Issue by Schwammerl11** [AvyAddons/BlizzardBars_Mouseover#4]
### Description

Flickering means every bar quickly fading in and out with different bars being active at the same time.
Only happens with the "Link Action Bars" option enabled, without the addon works fine.
Checked with different fade in/out delays and durations, made no difference.

### Addon Version

v2.1.0

### World of Warcraft Flavor

Retail

### Not sure this is regarding this error
```
_3x FadeBars/fade.lua:306: attempt to index global 'MainMenuBarArtFrame' (a nil value)
[string "@FadeBars/fade.lua"]:306: in main chunk_

Locals:
_ = "FadeBars"
core = <table> {
 Config = <table> {
 }
 init = <function> defined @FadeBars/init.lua:68
 commands = <table> {
 }
 Anim = <table> {
 }
 Print = <function> defined @FadeBars/init.lua:61
 Fade = <table> {
 }
}
Fade = <table> {
 updateCurrentVariables = <function> defined @FadeBars/fade.lua:190
 setupVariables = <function> defined @FadeBars/fade.lua:36
 combatVariables = <function> defined @FadeBars/fade.lua:98
 endcombatVariables = <function> defined @FadeBars/fade.lua:118
}
incombat = false
currentfadeopacity = 0.100000
currenthoveropacity = 1
currentmenubaropacity = 1
hoveringmainbar = false
hoveringleft = false
hoveringright = false
SpellFlyout = SpellFlyout {
 0 = <userdata>
 SetFixedWidth = <function> defined @SharedXML/LayoutFrame.lua:135
 MarkDirty = <function> defined @SharedXML/LayoutFrame.lua:99
 eventsRegistered = false
 AddLayoutChildren = <function> defined @SharedXML/LayoutFrame.lua:55
 SetFixedHeight = <function> defined @SharedXML/LayoutFrame.lua:139
 OnCleaned = <function> defined @SharedXML/LayoutFrame.lua:131
 SetBorderSize = <function> defined @FrameXML/SpellFlyout.lua:451
 SetFixedSize = <function> defined @SharedXML/LayoutFrame.lua:143
 GetLayoutChildren = <function> defined @SharedXML/LayoutFrame.lua:73
 OnUpdate = <function> defined @SharedXML/LayoutFrame.lua:93
 GetAdditionalRegions = <function> defined @SharedXML/LayoutFrame.lua:85
 GetFixedSize = <function> defined @SharedXML/LayoutFrame.lua:156
 SetHeightPadding = <function> defined @SharedXML/LayoutFrame.lua:160
 IsDirty = <function> defined @SharedXML/LayoutFrame.lua:127
 GetHeightPadding = <function> defined @SharedXML/LayoutFrame.lua:164
 Toggle = <function> defined @FrameXML/SpellFlyout.lua:206
 Layout = <function> defined @SharedXML/LayoutFrame.lua:395
 Background = Frame {
 }
 SetBorderColor = <function> defined @FrameXML/SpellFlyout.lua:444
 GetFixedHeight = <function> defined @SharedXML/LayoutFrame.lua:152
 GetWidthPadding = <function> defined @SharedXML/LayoutFrame.lua:168
 GetFixedWidth = <function> defined @SharedXML/LayoutFrame.lua:148
 OnShow = <function> defined @SharedXML/LayoutFrame.lua:32
 MarkIgnoreInLayout = <function> defined @SharedXML/LayoutFrame.lua:44
 MarkClean = <function> defined @SharedXML/LayoutFrame.lua:117
 IsLayoutFrame = <function> defined @SharedXML/LayoutFrame.lua:36
 IgnoreLayoutIndex = <function> defined @SharedXML/LayoutFrame.lua:391
}
FlyoutShown = false
FlyoutName = "none"
FlyoutUpdated = false
StatusTrackingBarNames = <table> {
}
StatusLoaded = false
showbar1 = <function> defined @FadeBars/fade.lua:221
hidebar1 = <function> defined @FadeBars/fade.lua:226
showbar2 = <function> defined @FadeBars/fade.lua:251
hidebar2 = <function> defined @FadeBars/fade.lua:256
showmenubar1 = <function> defined @FadeBars/fade.lua:278
hidemenubar1 = <function> defined @FadeBars/fade.lua:283
(*temporary) = nil
(*temporary) = nil
(*temporary) = "OnLeave"
(*temporary) = <function> defined @FadeBars/fade.lua:283
(*temporary) = true
(*temporary) = nil
(*temporary) = true
(*temporary) = nil
(*temporary) = "attempt to index global 'MainMenuBarArtFrame' (a nil value)"
```

### Reproduction Steps

1. Have "Link Action Bars" option enabled and UI reloaded.
2. Hover over one bar to make them visible.
3. Move your mouse over another bar.
- Sometimes it may take a little bit to bug out, but I got the flickering on every try.
- Also sometimes the flickering happens with just hovering over one bar for a short duration and then leaving the bar without hovering over another one.

### Screenshots

https://github.com/AvyAddons/BlizzardBars_Mouseover/assets/15922280/609d688b-f1eb-4886-a867-1f5ab411804a

### Tested with only BlizzardBars_Mouseover

- [X] Yes

**Comment by lucasvienna** [comment]
Thanks for the report, I'll investigate and get back to you soon!

**Comment by lucasvienna** [comment]
I've released [v2.1.1](https://github.com/AvyAddons/BlizzardBars_Mouseover/releases/tag/v2.1.1) with a tentative fix, it should propagate to Curse and Wago in the next hour or so.
Would you mind giving it a spin and see if that resolved the issue for you as well?

**Comment by Schwammerl11** [comment]
I tried really hard and with different fade out delays to recreate the bug, but it seems to be fixed!
Good job!
